### PR TITLE
Sudoku 4 suits

### LINKED ISSUE #2861: Sudoku with 4 Suits
I thought about it (and talked a bit with Hoddie), and I think we can make sense of 4-suit variants of Sudoku. The idea would be that we only have ranks 1,...,4 for each suit and then wrap around from 4 to 1.
The deck gets pretty small then (only 4 * 4 * 2 = 32 cards), so the pace/efficiency values would be:

- 3p: pace 4, efficiency 1.06
- 4p: pace 4, efficiency 1.14
- 5p: pace 1, efficiency 1.45
- 6p: pace 4, efficiency 1.14

(All assuming no dark suits, dark suits are just not winnable)

While 5p is certainly challenging, I think it's an interesting idea similar to 3-suit variants.

Would you accept a PR for these?

### PR DESCRIPTION
**Pull request by kesslermaximilian**
Fixes #2861.
Adds 4-suit variants of Sudoku as discussed there.

### CODE REVIEW

File: packages/client/src/game/ui/reactive/view/cardLayoutView.ts [hunk 1]
```diff
@@ -132,7 +132,13 @@ export function onPlayStacksChanged(
 
   if (variantRules.isSudoku(globals.variant)) {
     // First, we will find out all available stack starts.
-    const availableStackStartsFlags: boolean[] = [true, true, true, true, true];
+    const availableStackStartsFlags: boolean[] = new Array<boolean>(
+      globals.variant.ranks.length,
+    );
+    console.log(availableStackStartsFlags);
```
Comment (Zamiell) [inline packages/client/src/game/ui/reactive/view/cardLayoutView.ts:138]: .

Comment (kesslermaximilian) [reply packages/client/src/game/ui/reactive/view/cardLayoutView.ts:138]: Oh yeah, that should not be there.

Comment (kesslermaximilian) [reply packages/client/src/game/ui/reactive/view/cardLayoutView.ts]: it's removed now.

**Comment by Zamiell** [comment]
I'll merge this on August 30th or so

**Comment by Zamiell** [comment]
can you rebase this on main

**Comment by kesslermaximilian** [comment]
In main, you now have two identical functions `initArray` and `newArray`, I think we should remove one.

**Comment by kesslermaximilian** [comment]
Regarding the remaining linter errors here, I don't know what exactly those are and I think they're not caused by this PR...?

**Comment by Zamiell** [comment]
Oh yeah I'll get rid of initarray in favor of newarray

**Comment by Zamiell** [comment]
If lint passes on master then they have to be because of your pr

**Comment by Zamiell** [comment]
if you look at the pr you can see you are removing eslint disable comments in the files changed tab

**Comment by Zamiell** [comment]
if you rebase the pr on main again maybe the errors will go away

**Comment by kesslermaximilian** [comment]
I'll try to do so the next days, not sure when exactly I'll have time.

**Pull request merged**